### PR TITLE
CoderRegistry: make deprecated method private

### PR DESCRIPTION
**Pull request by dhalperi**
This was part of the public API signature in Dataflow, but need not be in Beam.

Also remove a few unneeded publics in the same file.

Also update CoderProvider javadoc

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java [hunk 1]
```diff
@@ -370,7 +370,7 @@ public void registerCoder(Class<?> clazz, CoderFactory coderFactory) {
    *
    * <p>See {@link #getFallbackCoderProvider}.
    */
-  public void setFallbackCoderProvider(CoderProvider coderProvider) {
+  void setFallbackCoderProvider(CoderProvider coderProvider) {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java]: This one is definitely public. It enables users to set various universal fallbacks beyond those that we configure when we ship.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java [hunk 2]
```diff
@@ -379,7 +379,7 @@ public void setFallbackCoderProvider(CoderProvider coderProvider) {
    *
    * <p>See {@link #setFallbackCoderProvider}.
    */
-  public CoderProvider getFallbackCoderProvider() {
+  private CoderProvider getFallbackCoderProvider() {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java]: And this one needs to be public so they can combine it with the above. You could replace them both with `addFallbackCoderProvider` so they are only allowed to add alternatives (which should have precedence)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java [hunk 3]
```diff
@@ -535,13 +532,13 @@ public CoderProvider getFallbackCoderProvider() {
     private Coder<?> coder;
     private Type type;
 
-    public IncompatibleCoderException(String message, Coder<?> coder, Type type) {
+    IncompatibleCoderException(String message, Coder<?> coder, Type type) {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java]: This is fine, but redundant and IMO not quite best practice. It means that if one refactors the class to have a different access level, one also has to scrape through the methods and change them.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java [hunk 4]
```diff
@@ -413,11 +413,8 @@ public CoderProvider getFallbackCoderProvider() {
    * @param baseClass the base type, a parameterized class
    * @param knownCoders a map corresponding to the set of known {@link Coder Coders} indexed by
    * parameter name
-   *
-   * @deprecated this method is not part of the public interface and will be made private
    */
-  @Deprecated
-  public <T> Map<Type, Coder<?>> getDefaultCoders(
+  private <T> Map<Type, Coder<?>> getDefaultCoders(
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java]: +1. And probably some related methods, like the one that takes an array...

**Comment by dhalperi** [comment]
PTAL. Also moved the //////// that separates private and public ;)

**Comment by kennknowles** [comment]
LGTM

**Pull request closed without merging**